src/main/java/duke/commands/UnmarkTaskCommand.java

```diff
@@ -0,0 +1,56 @@
+package duke.commands;
+
+import duke.exceptions.DukeException;
+import duke.managers.TaskManager;
+import duke.managers.UiManager;
+import duke.models.task.Task;
+
+/**
+ * Encapsulate a command for marking a {@link Task} as not completed. The command should be used as follows:
+ * <ul>
+ *     <li>
+ *         {@code unmark {taskNumber}}: To mark the task corresponding to the task number as not completed.
+ *     </li>
+ * </ul>
+ *
+ * @author Emily Ong Hui Qi
+ */
+public class UnmarkTaskCommand implements Command {
+    public static final String COMMAND_WORD = "unmark";
+
+    private static final String MESSAGE_MARK_TASK_AS_UNDONE = "OK, I've marked this task as not done yet:";
+
+    private static final String ERROR_MISSING_TASK_INDEX = "You are missing a task number!\n"
+            + "Use the 'list' command to view the tasks and their number.";
+    private static final String ERROR_NAN_TASK_NUMBER = "The task number you provided is not a number!";
```

Comment:
I liked how the commands generally have many constants (that follow the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#naming:~:text=Constant%20names%20must%20be%20all%20uppercase%20using%20underscore%20to%20separate%20words.)). The abstraction makes the code feel very clean.